lib/logging/logging.dart

```diff
@@ -114,158 +126,161 @@ class LoggingScreen extends Screen {
+        e.timestamp,
+        summary: e.json.toString(),
+      ));
+    }
+  }
 
-    // Log Flutter frame events.
-    service.onExtensionEvent.listen((Event e) {
-      if (e.extensionKind == 'Flutter.Frame') {
-        final FrameInfo frame = FrameInfo.from(e.extensionData.data);
-
-        final String frameId = '#${frame.number}';
-        final String frameInfo =
-            '<span class="pre">$frameId ${frame.elapsedMs.toStringAsFixed(1).padLeft(4)}ms </span>';
-        final String div = createFrameDivHtml(frame);
-
-        _log(LogData(
-          '${e.extensionKind.toLowerCase()}',
-          jsonEncode(e.extensionData.data),
-          e.timestamp,
-          summaryHtml: '$frameInfo$div',
-        ));
-      } else {
-        _log(LogData(
-          '${e.extensionKind.toLowerCase()}',
-          jsonEncode(e.json),
-          e.timestamp,
-          summary: e.json.toString(),
-        ));
-      }
-    });
+  void _handleGCEvent(Event e) {
+    final HeapSpace newSpace = HeapSpace.parse(e.json['new']);
+    final HeapSpace oldSpace = HeapSpace.parse(e.json['old']);
+    final Map<dynamic, dynamic> isolateRef = e.json['isolate'];
+
+    final int usedBytes = newSpace.used + oldSpace.used;
+    final int capacityBytes = newSpace.capacity + oldSpace.capacity;
+
+    final int time = ((newSpace.time + oldSpace.time) * 1000).round();
+
+    final String summary = '${isolateRef['name']} • '
+        '${e.json['reason']} collection in $time ms • '
+        '${printMb(usedBytes)} MB used of ${printMb(capacityBytes)} MB';
+
+    final Map<String, dynamic> event = <String, dynamic>{
+      'reason': e.json['reason'],
+      'new': newSpace.json,
+      'old': oldSpace.json,
+      'isolate': isolateRef,
+    };
+
+    final String message = jsonEncode(event);
+    _log(LogData('gc', message, e.timestamp, summary: summary));
+  }
+
+  void _handleDeveloperLogEvent(Event e) {
+    final VmServiceWrapper service = serviceManager.service;
+
+    final dynamic logRecord = e.json['logRecord'];
+
+    String loggerName =
+        _valueAsString(InstanceRef.parse(logRecord['loggerName']));
+    if (loggerName == null || loggerName.isEmpty) {
+      loggerName = 'log';
+    }
+    final int level = logRecord['level'];
+    final InstanceRef messageRef = InstanceRef.parse(logRecord['message']);
+    String summary = _valueAsString(messageRef);
+    if (messageRef.valueAsStringIsTruncated == true) {
+      summary += '...';
+    }
+    final InstanceRef error = InstanceRef.parse(logRecord['error']);
+    final InstanceRef stackTrace = InstanceRef.parse(logRecord['stackTrace']);
+
+    final String details = summary;
+    Future<String> detailsComputer;
+
+    // If the message string was truncated by the VM, or the error object or
+    // stackTrace objects were non-null, we need to ask the VM for more
+    // information in order to render the log entry. We do this asynchronously
+    // on-demand using the `detailsComputer` Future.
```

Comment:
Both backticks and square brackets only get interpreted in dartdoc comments (`///`), not regular line comments; I'm not sure there's a strong convention for which to use in line comments. I tend to avoid square brackets, as that implies that they're command-clickable, but the analysis server doesn't try and resolve them, so they don't actually link to anything.

I could remove the back-ticks here; no strong opinions.